shared-libs/contacts/src/places.js

```diff
@@ -83,23 +83,28 @@ const validatePlace = place => {
   return Promise.resolve();
 };
 
-const createPlace = place => {
-  const self = module.exports;
-  return self._validatePlace(place)
-    .then(() => {
-      const date = place.reported_date ? utils.parseDate(place.reported_date) : new Date();
-      place.reported_date = date.valueOf();
-      if (place.parent) {
-        place.parent = lineage.minifyLineage(place.parent);
-      }
-      if (place.contact) {
-        // also validates contact if creating
-        return people.getOrCreatePerson(place.contact).then(person => {
-          place.contact = lineage.minifyLineage(person);
-        });
-      }
-    })
-    .then(() => db.medic.post(place));
+const createPlace = async (place) => {
+  const contact = place.contact;
+  delete place.contact;
+
+  await module.exports._validatePlace(place);
+
+  const date = place.reported_date ? utils.parseDate(place.reported_date) : new Date();
+  place.reported_date = date.valueOf();
+  if (place.parent) {
+    place.parent = lineage.minifyLineage(place.parent);
+  }
+
+  const response = await db.medic.post(place);
+  if (!contact) {
+    return response;
+  }
+  const placeUUID = response.id;
+
+  contact.place = placeUUID;
+  const person = await people.getOrCreatePerson(contact);
+  const result = await updatePlace(placeUUID, { contact: person._id });
+  return { ...result, contact: { id: person._id, rev: person._rev } };
```

Comment:
No the rev isn't used anywhere but it's included in the response for place, so i decided to also include it in contact

Reply:
Cool, I think there is a slight risk to changing the API response without versioning though, so if this is not used I think it's safer not to include it - in case integrations require specific responses.